wallet/src/signer/ledger_signer/mod.rs

```diff
@@ -0,0 +1,1306 @@
+                        sign_classical_multisig_spending, AuthorizedClassicalMultisigSpend,
+                        ClassicalMultisigCompletionStatus,
+                    },
+                    multisig_partial_signature::{self, PartiallySignedMultisigChallenge},
+                },
+                standard_signature::StandardInputSignature,
+                InputWitness,
+            },
+            sighash::{sighashtype::SigHashType, signature_hash},
+            DestinationSigError,
+        },
+        AccountCommand, ChainConfig, Destination, OrderAccountCommand, SignedTransactionIntent,
+        Transaction, TxInput, TxOutput,
+    },
+    primitives::{BlockHeight, Idable, H256},
+};
+use crypto::key::{
+    extended::ExtendedPublicKey,
+    hdkd::{derivable::Derivable, u31::U31},
+    signature::SignatureKind,
+    PrivateKey, SigAuxDataProvider, Signature, SignatureError,
+};
+use serialization::Encode;
+use utils::ensure;
+use wallet_storage::{
+    WalletStorageReadLocked, WalletStorageReadUnlocked, WalletStorageWriteUnlocked,
+};
+use wallet_types::{
+    account_info::DEFAULT_ACCOUNT_INDEX,
+    hw_data::{HardwareWalletFullInfo, LedgerData, LedgerFullInfo, LedgerModel},
+    partially_signed_transaction::{
+        PartiallySignedTransaction, PtxAdditionalInfo, TokensAdditionalInfo,
+    },
+    signature_status::SignatureStatus,
+    AccountId,
+};
+
+use async_trait::async_trait;
+use itertools::{izip, Itertools};
+use ledger_lib::{info::Model, Exchange, Filters, LedgerHandle, LedgerProvider, Transport};
+use mintlayer_ledger_messages::{
+    AdditionalUtxoInfo, AddrType, Bip32Path as LedgerBip32Path, CoinType,
+    InputAddressPath as LedgerInputAddressPath, SighashInputCommitment as LSighashInputCommitment,
+    Signature as LedgerSignature, TxInputReq, TxInputWithAdditionalInfo, TxOutputReq,
+};
+use randomness::make_true_rng;
+use tokio::sync::Mutex;
+
+/// Signer errors
+#[derive(thiserror::Error, Debug, Eq, PartialEq)]
+pub enum LedgerError {
+    #[error("No connected Ledger device found")]
+    NoDeviceFound,
+    #[error("Connected to an unknown Ledger device model")]
+    UnknownModel,
+    #[error("Device timeout")]
+    DeviceTimeout,
+    #[error("A different app is currently open on your Ledger device: \"{0}\". Please close it and open the Mintlayer app instead.")]
+    DifferentActiveApp(String),
+    #[error("Received an invalid response from the Ledger device")]
+    InvalidResponse,
+    #[error("Received an error response from the Ledger device: {0}")]
+    ErrorResponse(String),
+    #[error("Device error: {0}")]
+    DeviceError(String),
+    #[error("Missing hardware wallet data in database")]
+    MissingHardwareWalletData,
+    #[error("Derivation path is too long to send to Ledger")]
+    PathToLong,
+    #[error("Invalid public key returned from Ledger")]
+    InvalidKey,
+    #[error("The file being loaded is a software wallet and does not correspond to the connected hardware wallet")]
```

Comment:
Same comment about the "and does not correspond to the connected hardware wallet" part.